pkg/processor/trigger/kafka/trigger.go

```diff
@@ -360,6 +359,7 @@ func (k *kafka) drainOnRebalance(session sarama.ConsumerGroupSession,
 
 		wg.Wait()
 		readyForRebalanceChan <- true
+		close(readyForRebalanceChan)
```

Comment:
In that case if we are out of waiting time (in select operator), we will never close the chan because writing to the chan is blocking operation. So, I would leave it as is.

Reply:
The goroutine will still run even when the timeout has passed and the select doesn't wait for this channel.

The issue is that if timeout has passed, the function exists and closes the `readyForRebalanceChan`. The goroutine continues to run, and tries to write to the closed channel.
My thought was to not close the channel before the goroutine is done.

Reply:
@TomerShor Yes, but if the goroutine is still running when the timeout has passed, this change will result in the channel never being closed. We won't read from readyForRebalanceChan, leading to a zombie goroutine persisting indefinitely.

Currently, it's possible to attempt writing to a closed channel, causing a panic and the goroutine to exit. To address this issue properly and prevent the panic, we should notify the goroutine from the main function body that the timeout has passed, and there's no need to write anything to the channel. But as for me, panicing in this goroutine is not a big issue

Reply:
@rokatyy We already have a recover for that case in this goroutine 🤦 
I will revert this change.